packages/metadata-cmp/src/compare.ts

```diff
@@ -91,6 +100,91 @@ async function main (): Promise<number> {
+        .forEach((c): void => {
+          const cA = decA.query[n][c];
+          const cB = decB.query[n][c];
+
+          // storage types differ
+          if (!cA.meta.type.eq(cB.meta.type)) {
+            // diff map
+            if (cA.meta.type.isMap && cB.meta.type.isMap) {
+              const mapA = cA.meta.type.asMap;
+              const mapB = cB.meta.type.asMap;
+              const diffs = [];
+
+              if (!mapA.hasher.eq(mapB.hasher)) {
+                diffs.push(createCompare('hasher', '', mapA.hasher.toString(), mapB.hasher.toString(), undefined, true));
+              }
+
+              if (!mapA.key.eq(mapB.key)) {
+                diffs.push(createCompare('key', '', mapA.key.toString(), mapB.key.toString(), undefined, true));
+              }
+
+              if (!mapA.value.eq(mapB.value)) {
+                diffs.push(createCompare('value', '', mapA.value.toString(), mapB.value.toString(), undefined, true));
+              }
+
+              if (diffs.length > 0) {
+                console.log(createLog(c, diffs.shift() || '', ''));
+
+                for (const diff of diffs) {
+                  console.log(createLog('', diff, ''));
+                }
+              }
+            } else if (cA.meta.type.isDoubleMap && cB.meta.type.isDoubleMap) {
+              // diff double map
+              const mapA = cA.meta.type.asDoubleMap;
+              const mapB = cB.meta.type.asDoubleMap;
+              const diffs = [];
+
+              if (!mapA.hasher.eq(mapB.hasher)) {
+                diffs.push(createCompare('hasher', '', mapA.hasher.toString(), mapB.hasher.toString(), undefined, true));
+              }
+
+              if (!mapA.key1.eq(mapB.key1)) {
+                diffs.push(createCompare('key1', '', mapA.key1.toString(), mapB.key1.toString(), undefined, true));
+              }
+
+              if (!mapA.key2Hasher.eq(mapB.key2Hasher)) {
+                diffs.push(createCompare('key2Hasher', '', mapA.key2Hasher.toString(), mapB.key2Hasher.toString(), undefined, true));
+              }
+
+              if (!mapA.key2.eq(mapB.key2)) {
+                diffs.push(createCompare('key2', '', mapA.key2.toString(), mapB.key2.toString(), undefined, true));
+              }
+
+              if (!mapA.value.eq(mapB.value)) {
+                diffs.push(createCompare('value', '', mapA.value.toString(), mapB.value.toString(), undefined, true));
+              }
+
+              if (diffs.length > 0) {
+                console.log(createLog(c, diffs.shift() || '', ''));
+
+                for (const diff of diffs) {
+                  console.log(createLog('', diff, ''));
+                }
+              }
+            } else if (cA.meta.type.isPlain && cB.meta.type.isPlain) {
+              // diff plain type
+              const tA = cA.meta.type.asPlain;
+              const tB = cB.meta.type.asPlain;
+              console.log(createCompare(c, 'type', tA.toString(), tB.toString(), undefined));
```

Comment:
```suggestion
              const tB = cB.meta.type.asPlain;

              console.log(createCompare(c, 'type', tA.toString(), tB.toString(), undefined));
```

Last one, not sure why I missed it (the linter did complain) in the last batch.